No suitable Python interpreter found. Windows 10 Command Prompt

Help! I get this error!!!

```
| |  \____ \|  ___)    | |    | |   / _ \|  _ \| ___ |  _ \(____ |/ _  | ___ |
| |_ _____) ) |        | |   _| |  | |_| | |_| | ____| | | / ___ ( (_| | ____|
|___|______/|_|        |_|  (___|   \___/|  __/|_____)_| |_\_____|\___ |_____)
                                         |_|                     (_____|

Welcome to the SFT technologies computer-aided openage build system!

You have chosen, or been chosen, to attempt the daring task of building openage.
If you have installed all the dependencies that are conveniently listed in
[doc/building.md], this _might_ just work!

If it doesn't, consider reporting the issue/asking for help in #sfttech on freenode.net.

CMake Warning at buildsystem/HandleCXXOptions.cmake:82 (message):
  Using untested compiler, at least I hope it's free software.  Continue on
  your own, warrior.
Call Stack (most recent call first):
  CMakeLists.txt:111 (include)


No suitable Python interpreter found.
We need an interpreter that is shipped with libpython and header files.
Specify your own with -DPYTHON=/path/to/executable



CMake Error at cmake-3.10.0-win32-x86/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Python (missing: PYTHON PYTHON_INCLUDE_DIRS
  PYTHON_LIBRARIES) (Required is at least version "3.4")
Call Stack (most recent call first):
  cmake-3.10.0-win32-x86/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  buildsystem/modules/FindPython.cmake:231 (find_package_handle_standard_args)
  C:/Users/Ryan/Desktop/VCPKG/vcpkg-master/scripts/buildsystems/vcpkg.cmake:211 (_find_package)
  buildsystem/HandlePythonOptions.cmake:6 (find_package)
  CMakeLists.txt:113 (include)


-- Configuring incomplete, errors occurred!
See also "C:/Users/Ryan/Desktop/SFTtech-openage-fb4a9eb/build/CMakeFiles/CMakeOutput.log".

C:\Users\Ryan\Desktop\SFTtech-openage-fb4a9eb\build>
```


How do I fix it??!!

Now, on the thing I had first, I get 

LINK : fatal error LNK1104: cannot open file 'python36_d.lib' [C:\Users\Ryan\Desktop\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_54716.vcxproj

And I still get 

"C:\Users\Ryan\Desktop\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_54716.vcxproj" (default target) (1) ->
(Link target) ->
  LINK : fatal error LNK1104: cannot open file 'python36_d.lib' [C:\Users\Ryan\Desktop\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_54716.vcxproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:03.17



No suitable Python interpreter found.
We need an interpreter that is shipped with libpython and header files.
Specify your own with -DPYTHON=/path/to/executable



CMake Error at cmake-3.10.0-win32-x86/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Python (missing: PYTHON PYTHON_INCLUDE_DIRS
  PYTHON_LIBRARIES) (Required is at least version "3.4")
Call Stack (most recent call first):
  cmake-3.10.0-win32-x86/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  buildsystem/modules/FindPython.cmake:231 (find_package_handle_standard_args)
  C:/Users/Ryan/Desktop/VCPKG/vcpkg-master/scripts/buildsystems/vcpkg.cmake:211 (_find_package)
  buildsystem/HandlePythonOptions.cmake:6 (find_package)
  CMakeLists.txt:113 (include)


-- Configuring incomplete, errors occurred!
See also "C:/Users/Ryan/Desktop/SFTtech-openage-fb4a9eb/build/CMakeFiles/CMakeOutput.log".

I have everything working now

EXCEPT THIS ERROR::::

  FindPython_test.obj : error LNK2019: unresolved external symbol __imp__Py_Initialize referenced in function _main [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
  FindPython_test.obj : error LNK2019: unresolved external symbol __imp__Py_Finalize referenced in function _main [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
  C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\Debug\cmTC_15a81.exe : fatal error LNK1120: 2 unresolved externals [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]

All programs are working, but the error::


  FindPython_test.obj : error LNK2019: unresolved external symbol __imp__Py_Initialize referenced in function _main [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
  FindPython_test.obj : error LNK2019: unresolved external symbol __imp__Py_Finalize referenced in function _main [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
  C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\Debug\cmTC_15a81.exe : fatal error LNK1120: 2 unresolved externals [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]

Sorry, I did not see the beginning of your last comment...

'''
  FindPython_test.obj : error LNK2019: unresolved external symbol __imp__Py_Initialize referenced in function _main [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
  FindPython_test.obj : error LNK2019: unresolved external symbol __imp__Py_Finalize referenced in function _main [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
  C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\Debug\cmTC_15a81.exe : fatal error LNK1120: 2 unresolved externals [C:\SFTtech-openage-fb4a9eb\build\CMakeFiles\CMakeTmp\cmTC_15a81.vcxproj]
'''

Ok, I am getting a new error:

```
-- Using python interpreter: C:/Program Files (x86)/Python36-32/python.exe
Traceback (most recent call last):
  File "C:\Program Files (x86)\Python36-32\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files (x86)\Python36-32\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\SFTtech-openage-fb4a9eb\openage\__main__.py", line 133, in <module>
    sys.exit(main())
  File "C:\SFTtech-openage-fb4a9eb\openage\__main__.py", line 83, in main
    from .convert.main import init_subparser
  File "C:\SFTtech-openage-fb4a9eb\openage\convert\main.py", line 7, in <module>
    import readline  # pylint: disable=unused-import
ModuleNotFoundError: No module named 'readline'
CMake Error at buildsystem/codegen.cmake:25 (message):
  failed to get target list from codegen invocation
Call Stack (most recent call first):
  libopenage/CMakeLists.txt:54 (codegen_run)


-- Configuring incomplete, errors occurred!
See also "C:/SFTtech-openage-fb4a9eb/build/CMakeFiles/CMakeOutput.log".
```

--- a/buildsystem/HandleCXXOptions.cmake
+++ b/buildsystem/HandleCXXOptions.cmake
@@ -67,7 +67,11 @@ elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
 		set_compiler_flags("CXX" "-stdlib=libc++")
 	endif()
 
-else() #"Intel", "MSVC", etc..
+elseif(MSVC)
+	# Don't worry. You're not alone. If you face an issue, just ask.
+	return() # The following flag specifications don't apply.
+
+else() # "Intel", etc..
 	message(WARNING "Using untested compiler, at least I hope it's free software. Continue on your own, warrior.")
 endif()
 

--- a/doc/build_instructions/windows_msvc.md
+++ b/doc/build_instructions/windows_msvc.md
@@ -10,13 +10,14 @@
  - [Python 3](https://www.python.org/downloads/windows/)
    - With the "pip" option enabled. We use `pip` to install other dependencies.
    - With the "Precompile standard library" option enabled.
+   - With the "Download debug binaries (...)" option enabled.
    - If in doubt, run the installer again and choose "Modify".
  - [CMake](https://cmake.org/download/)
 
 ### Python Modules
  Open a command prompt at `<Python 3 installation directory>/Scripts`
 
-    pip install numpy pillow pygments cython
+    pip install cython numpy pillow pygments pyreadline
 
 ### vcpkg packages
  Set up [vcpkg](https://github.com/Microsoft/vcpkg#quick-start). Open a command prompt at `<vcpkg directory>`

--- a/doc/build_instructions/windows_msvc.md
+++ b/doc/build_instructions/windows_msvc.md
@@ -1,9 +1,11 @@
 # Procedure for Microsoft Windows users (Windows 10)
 
- Since Windows doesn't offer a native package manager, we use a mixture of manual and automated steps to get the dependencies for openage. *Please remember to replace the directories referenced below (written in <...>) with the appropriate values.*
+ Since Windows doesn't offer a native package manager, we use a mixture of manual and automated steps to get the dependencies for openage.
+ *Please remember to replace the directories referenced below (written in <...>) with the appropriate values.*
 
 ## Setting up the build environment
- You will need to download and install the following manually. Those who already have the latest stable versions of these softwares can skip this:
+ You will need to download and install the following manually.
+ Those who already have the latest stable versions of these softwares can skip this:
  - [Visual Studio 2017 Community edition](https://www.visualstudio.com/downloads/)
    - With the "Desktop development with C++" workload.
    - With the "Windows 10 SDK". Choose the latest version listed.
@@ -24,7 +26,9 @@
 
     vcpkg install dirent libepoxy fontconfig freetype harfbuzz opus opusfile qt5 sdl2 sdl2-image libpng
 
- _Note:_ Building and installing `qt5` using vcpkg takes a lot of space-time. You might want to install [the prebuilt version](https://www.qt.io/download-open-source/) instead. This should be mentioned in `CMAKE_PREFIX_PATH` with the cmake command below.
+ _Note:_ Building and installing `qt5` using vcpkg takes a lot of space-time.
+ You might want to install [the prebuilt version](https://www.qt.io/download-open-source/) instead.
+ Include `-DCMAKE_PREFIX_PATH=<QT5 directory>` in the cmake configure command.
 
 ### The "other missing" dependencies
  **opus-tools (for `opusenc`)**
@@ -48,24 +52,22 @@
 
      mkdir build
      cd build
-     cmake -DCMAKE_TOOLCHAIN_FILE=<vcpkg directory>/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
+     cmake -DCMAKE_TOOLCHAIN_FILE=<vcpkg directory>/scripts/buildsystems/vcpkg.cmake ..
      cmake --build . --config RelWithDebInfo -- /nologo /m /v:m
 
 ## Running openage (in devmode)
  While this is straightforward on other platforms, there is still stuff to do to run openage on Windows:
-  - Install the [DejaVu Book Font](https://dejavu-fonts.github.io/Download.html). Download and extract the latest `dejavu-fonts-ttf` tarball/zip file.
-    - Install `ttf/DejaVuSerif*.ttf` font files. This can be done by double-clicking or copying those to `%WINDIR%/Fonts`.
+  - Install the [DejaVu Book Font](https://dejavu-fonts.github.io/Download.html).
+    - Download and extract the latest `dejavu-fonts-ttf` tarball/zip file.
+    - Copy `ttf/DejaVuSerif*.ttf` font files to `%WINDIR%/Fonts`.
     - Set the `FONTCONFIG_PATH` environment variable to `<vcpkg directory>\installed\<relevant config>\tools\fontconfig\fonts\conf.d`.
-    - Copy `fontconfig/57-dejavu-serif.conf` to the same.
+    - Copy `fontconfig/57-dejavu-serif.conf` to `%FONTCONFIG_PATH%`.
   - [Optional] Set the `AGE2DIR` environment variable to the AoE 2 installation directory.
-  - Copy the binaries from the `build` subdirectories to the required locations:
-
-        set ConfigToUse=RelWithDebInfo
-
-        copy libopenage\%ConfigToUse%\openage.dll ..\
-        copy libopenage\%ConfigToUse%\openage.pdb ..\
   - Append the following to the environment `PATH`:
+    - `<openage directory>\build\libopenage\<config built>` (for `openage.dll`)
+    - Path to `nyan.dll` (depends on the procedure chosen to get nyan)
     - `<vcpkg directory>\installed\<relevant config>\bin`
+    - `<QT5 directory>\bin` (if prebuilt QT5 was installed)
     - Path to `opusenc.exe`
 
  Now, execute `<openage directory>/run.exe` and enjoy!

--- a/doc/build_instructions/windows_msvc.md
+++ b/doc/build_instructions/windows_msvc.md
@@ -53,7 +53,8 @@
 
 ## Running openage (in devmode)
  While this is straightforward on other platforms, there is still stuff to do to run openage on Windows:
-  - Install the [DejaVu Book Font](https://dejavu-fonts.github.io/Download.html).
+  - Install the [DejaVu Book Font](https://dejavu-fonts.github.io/Download.html). Download and extract the latest `dejavu-fonts-ttf` tarball/zip file.
+    - Install `ttf/DejaVuSerif*.ttf` font files. This can be done by double-clicking or copying those to `%WINDIR%/Fonts`.
     - Set the `FONTCONFIG_PATH` environment variable to `<vcpkg directory>\installed\<relevant config>\tools\fontconfig\fonts\conf.d`.
     - Copy `fontconfig/57-dejavu-serif.conf` to the same.
   - [Optional] Set the `AGE2DIR` environment variable to the AoE 2 installation directory.

